api/integrations/launch_darkly/services.py

```diff
@@ -126,11 +635,30 @@ def _create_boolean_feature_states(
             feature_state=feature_state,
         )
 
+        _import_targets(
+            import_request=import_request,
+            ld_flag_config=ld_flag_config,
+            feature=feature,
+            environment=environment,
+            segments_by_ld_key=segments_by_ld_key,
+            mv_feature_options_by_variation={},
+        )
+        _import_rules(
+            import_request=import_request,
+            ld_flag_config=ld_flag_config,
+            feature=feature,
+            environment=environment,
+            segments_by_ld_key=segments_by_ld_key,
+            mv_feature_options_by_variation={},
+        )
```

Comment:
I don't like that we're introducing side effects here. We should consider one of the following:

- Factor out `_import_targets` and `_import_rules` calls so they're invoked directly from `process_import_request` rather than as a side effect. Make the factory functions return created entities, including the MV options, so they can be used by later code, or think of some common context object that can be used throughout the import process.
- Rename factory functions to `_create_boolean_feature_states_with_segments_identities` and so on.

Reply:
Hmm why you think this is a side-effect? Isn't targeting rules a part of the feature flag itself? 

I am not clear on how we are going to achieve this 🤔 As there is already lots of automated tests, can you push a commit that would do this refactoring? I feel like this  should not change how we create stuff.

Reply:
> Isn't targeting rules a part of the feature flag itself?

In Flagsmith world they aren't, rather, it's the other way around — feature states are referenced by segment/identity overrides and are otherwise oblivious to them.

> As there is already lots of automated tests, can you push a commit that would do this refactoring?

I'm off for 4 work days so it won't happen soon. As the change is mostly for better code structure/readability, I'm fine with having a TODO + an issue for this. Leaving this for @matthewelwell to sort out.

Reply:
Created an issue here: https://github.com/Flagsmith/flagsmith/issues/3383

Also added TODOs 👍